src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -0,0 +1,77 @@
+package wolox.training.controllers;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.exceptions.UserIdMismatchException;
+import wolox.training.exceptions.UserNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.models.User;
+import wolox.training.repositories.BookRepository;
+import wolox.training.repositories.UserRepository;
+
+@RestController
+@RequestMapping("/api/users")
+public class UserController {
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private BookRepository bookRepository;
+
+    @GetMapping("/{username}")
+    public User findByName(@PathVariable String username) {
+        User user = userRepository.findByUsername(username);
+        if (user == null) throw new UserNotFoundException();
+        return user;
+    }
+
+    @PostMapping
+    public User create(@RequestBody User user) {
+        return userRepository.save(user);
+    }
+
+    @PutMapping("/{id}")
+    public User update(@RequestBody User user, @PathVariable Long id) {
+        if (user.getId() != id) {
+            throw new UserIdMismatchException("The user id does not match");
+        }
+        if (!userRepository.existsById(id)) {
+            throw new UserNotFoundException();
+        }
+        return userRepository.save(user);
+    }
+
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
+        if (!userRepository.existsById(id)) {
+            throw new UserNotFoundException();
+        }
+        userRepository.deleteById(id);
```

Comment:
Same here,  you could do  `userRepository.deleteById(id).orElseThrow(...)`

Reply:
Cannot resolve method :(

Reply:
My bad 😅